server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/BazelProjectMapper.kt

```diff
@@ -88,7 +89,14 @@ class BazelProjectMapper(
         val librariesToImport = measure("Merge all libraries") {
             librariesFromDepsAndTargets + extraLibrariesFromJdeps.values.flatten().associateBy { it.label }
         }
-        return Project(workspaceRoot, modifiedModules.toList(), sourceToTarget, librariesToImport)
+        val rustExternalTargetsToImport = measure("Select external Rust targets") {
+            selectRustExternalTargetsToImport(rootTargets, dependencyTree)
+        }
+        val rustExternalModules = measure("Create Rust external modules") {
+            createRustExternalModules(rustExternalTargetsToImport, dependencyTree, librariesFromDeps)
+        }
+        val allModules = modifiedModules + rustExternalModules
```

Comment:
I don't think we can do that. `rustExternalModules` are later used in response to `RustWorkspaceRequest` in the same way as the "normal" modules are. This request requires information about all packages in the project (also external ones). Since Bazel doesn't have packages, we create them based on modules. That is why we need information stored in `Module` structure for `rustExternalModules` and cannot map them to `Library` structure that loses a lot of information